src/test/java/org/jenkinsci/backend/ircbot/IrcListenerTest.java

```diff
@@ -199,4 +207,143 @@ public Void answer(InvocationOnMock invocation) throws Throwable {
+
+        MessageEvent<?> event = mock(MessageEvent.class);
+        when(event.getChannel()).thenReturn(chan);
+        when(event.getMessage()).thenReturn("jenkins-admin: make "+newCommitter+" a committer on "+repoName);
+        when(event.getUser()).thenReturn(sender);
+
+        PircBotX bot = mock(PircBotX.class);
+        when(bot.getNick()).thenReturn("jenkins-admin");
+        when(event.getBot()).thenReturn(bot);
+
+        assertEquals(0, addedUsers.size());
+
+        ircListener.onMessage(event);
+
+        assertEquals(1, addedUsers.size());
+    }
+
+    public void testAddCommitters() throws Exception {
+        final String channel = "#jenkins";
+        final String botUser = "bot-user";
+        final List<String> newCommitters = new ArrayList<>();
+        newCommitters.add("new-committer1");
+        newCommitters.add("new-committer2");
+        newCommitters.add("new-committer3");
+
+        final List<GHUser> addedUsers = new ArrayList<>();
+
+        final String repoName = "the-repo";
+
+        PowerMockito.mockStatic(GitHub.class);
+
+        GitHub gh = mock(GitHub.class);
+        when(GitHub.connect()).thenReturn(gh);
+
+        GHOrganization o = mock(GHOrganization.class);
+        when(gh.getOrganization(IrcBotConfig.GITHUB_ORGANIZATION)).thenReturn(o);
+
+        GHRepository repo = mock(GHRepository.class);
+        when(repo.getName()).thenReturn(repoName);
+
+        when(o.getRepository(repoName)).thenReturn(repo);
+
+        GHTeam team = mock(GHTeam.class);
+        Map<String, GHTeam> teams = new HashMap<>();
+        teams.put(repoName+" Developers", team);
+        when(o.getTeams()).thenReturn(teams);
+
+        GHUser committerUser;
+        for(String newCommitter : newCommitters) {
+            committerUser = mock(GHUser.class);
+            when(gh.getUser(newCommitter)).thenReturn(committerUser);
+            doAnswer((Answer) invocation -> {
+                GHUser u = invocation.getArgumentAt(0, GHUser.class);
+                addedUsers.add(u);
+                return null;
+            }).when(team).add(committerUser);
+        }
+
+        System.setProperty("ircbot.testSuperUser", botUser);
+
+        IrcListener ircListener = new IrcListener(null);
+        User sender = mock(User.class);
+        when(sender.getNick()).thenReturn(botUser);
+
+        Channel chan = mock(Channel.class);
+        when(chan.getName()).thenReturn(channel);
+
+        OutputChannel out = mock(OutputChannel.class);
+        when(chan.send()).thenReturn(out);
+
+        ImmutableSortedSet.Builder<UserLevel> builder = ImmutableSortedSet.naturalOrder();
+        builder.add(UserLevel.VOICE);
+        when(sender.getUserLevels(chan)).thenReturn(builder.build());
+
+        MessageEvent<?> event = mock(MessageEvent.class);
+        when(event.getChannel()).thenReturn(chan);
+        when(event.getMessage()).thenReturn("jenkins-admin: make "+String.join(",  ", newCommitters)+" committers on "+repoName);

```

Comment:
I always find it nicer in tests to be explicit on what's been doing rather than generating where possible, it makes them much easier to read while making less DRY which isn't such a problem for test code.

I would suggest:
```suggestion
        when(event.getMessage()).thenReturn("jenkins-admin: make new-committer1, new-committer2, new-committer3 committers on " + repoName);
```